internal/kibana/alerting.go

```diff
@@ -21,6 +22,14 @@ var frequencyMinSupportedVersion = version.Must(version.NewVersion("8.6.0"))
 var alertsFilterMinSupportedVersion = version.Must(version.NewVersion("8.9.0"))
 var alertDelayMinSupportedVersion = version.Must(version.NewVersion("8.13.0"))
 
+// Avoid lint error on deprecated SchemaValidateFunc usage.
+//
+//nolint:staticcheck
+func stringIsAlertingDuration() schema.SchemaValidateFunc {
+	r := regexp.MustCompile(`^[0-9]+(?:\.[0-9]+)?(?:d|h|m|s)$`)
```

Comment:
Thanks for the change!

If I'm interpreting the regex correctly this block `(?:\.[0-9]+)?` is for decimals? That won't be necessary, the suggestion below [is more in tune with what we use as validation in our API](x-pack/plugins/alerting/common/routes/rule/validation/validate_duration/v1.ts).

```suggestion
	r := regexp.MustCompile(`^[1-9][0-9]*(?:d|h|m|s)$`)
```